aws-amplify/amplify-ios
# AsyncEvent refactoring final merge

*Description of changes:*

This PR rebases https://github.com/aws-amplify/amplify-ios/tree/palpatim/async-event-refactoring against master, and fixes a few stray issues:
- Removed unnecessary AWSDataStoreCategoryPluginTests.xcscheme
- Fix reference to awsconfiguration.json in AWSAuthPluginIntegrationTests
- Fix AWSAuthPluginTests formatting & linting
- Fix corrupted project file reference to AWSAuthSignUpOperation.swift
- Fix Analytics formatting & linting
- Change AuthDirective tests to use OperationResult instead of AsyncEvent
- Fix linting & formatting
- Fix corrupted project file reference to AuthResetPasswordRequest.swift

And importantly:
- Update CHANGELOG & prep for release


Although the base of this PR is https://github.com/aws-amplify/amplify-ios/tree/palpatim/async-event-refactoring, that is only to help isolate the changes for review. Once the PR is reviewed, I will change the base to https://github.com/aws-amplify/amplify-ios/tree/master and merge it. Therefore, it is safe to disregard conflicts with the "async-refactor" branch.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
